src/Interop/src/Hook.cpp

```diff
@@ -168,11 +168,11 @@ namespace OpenLoco::Interop
 
     void registerHook(uintptr_t address, hook_function function)
     {
+        size_t size = kMaxHooks * kHookByteCount;
         if (!_hookTableAddress)
         {
-            size_t size = kMaxHooks * kHookByteCount;
 #ifdef _WIN32
-            _hookTableAddress = VirtualAllocEx(GetCurrentProcess(), NULL, size, MEM_COMMIT, PAGE_EXECUTE_READWRITE);
+            _hookTableAddress = VirtualAllocEx(GetCurrentProcess(), NULL, size, MEM_COMMIT | MEM_RESERVE, PAGE_READWRITE);
```

Comment:
Do we definitely need to change this. We have executable code written into the hook table. This is causing DEP to trigger.